File: task.hs
module Task (sec, cot, csc, task) where


sec = (/) 1 . cos
csc = (/) 1 . sin

cot x = cos x / sin x


task :: Double -> Double
task x | x <= 0 = let secX = sec x in (((secX / cot x) ** 2) ** 3) - csc x - sec x
task x | x  > 0 = let
    log3 = log x / log 3
    log5 = log x / log 5
  in (((log5 / log5 * log3) ** 3) ** 3) / (log5 / log5 + log5 + log3)

File: src/main/kotlin/ru/byprogminer/lab2testing/math/task.kt
package ru.byprogminer.lab2testing.math

import kotlin.math.pow


fun task(
    sec: MathFunction,
    cot: MathFunction,
    csc: MathFunction,
    log: BasedMathFunction,
): MathFunction = { x ->
    if (x <= 0) {
        val secX = sec(x)

        (secX / cot(x)).pow(2).pow(3) - csc(x) - secX
    } else {
        val log3 = log(x, 3)
        val log5 = log(x, 5)

        (log5 / log5 * log3).pow(3).pow(3) / (log5 / log5 + log5 + log3)
    }
}
